src/components/organize/CampaignDetailsHeader.tsx

```diff
@@ -0,0 +1,156 @@
+                                        year="numeric"
+                                    />
+                                </>
+                            ) : (
+                                <Msg id="pages.organizeCampaigns.indefinite" />
+                            ) }
+                        </Typography>
+                    </Box>
+                    { /* Action Buttons */ }
+                    <Box alignItems="center" display="flex" mb={ 2 }>
+                        { /* Public Page Link */ }
+                        <Box mr={ 2 }>
+                            <NextLink href={ `/o/${orgId}/campaigns/${campId}` } passHref>
+                                <Button color="primary">
+                                    <Msg id="pages.organizeCampaigns.linkGroup.public"/>
+                                </Button>
+                            </NextLink>
+                        </Box>
+                        { /* Edit Campiagn Activator */ }
+                        <Button color="primary" onClick={ () => setEditCampaignDialogOpen(true) } variant="contained">
+                            <Msg id="pages.organizeCampaigns.linkGroup.settings"/>
+                        </Button>
+                    </Box>
+                </Box>
+                { /* Campaign Manager */ }
+                { campaign.manager ? (
+                    <NextLink href={ `/organize/${orgId}/people/${campaign.manager.id}` }>
+                        <Box alignItems="center" className={ classes.personContainer } display="flex">
+                            <Box mr={ 2 }>
+                                <Avatar
+                                    src={ `/api/orgs/${orgId}/people/${campaign.manager.id}/avatar` }
+                                />
+                            </Box>
+                            <Box display="flex" flexDirection="column">
+                                <Typography variant="h6">{ campaign.manager.name }</Typography>
+                                <Typography variant="body2"><Msg id="pages.organizeCampaigns.campaignManager" /></Typography>
+                            </Box>
+                        </Box>
+                    </NextLink>
+                ) :
+                    // If no campaign manager
+                    <Typography variant="body1">
+                        <Msg id="pages.organizeCampaigns.noCampaignManager" />
+                    </Typography>
+                }
+            </Box>
+
+            <ZetkinDialog
+                onClose={ () => setEditCampaignDialogOpen(false) }
+                open={ editCampaignDialogOpen }
+                title={ intl.formatMessage({ id: 'misc.formDialog.campaign.edit' }) }>
+                <CampaignForm
+                    campaign={ campaign }
+                    onCancel={ () => setEditCampaignDialogOpen(false) }
+                    onSubmit={ handleEditCampaignFormSubmit }
+                />
+            </ZetkinDialog>
```

Comment:
IMO it's a little bit confusing that the dialog is now in a specific component rather than at page-level. It makes sense in this context, but in many other contexts (e.g. events, tasks, etc) the edit dialog might more likely be a part of the page, and that is cause for inconsistency. This approach also makes it more difficult to open the dialog from any other place (e.g. from a link in the header or elsewhere).

What do you think of this?

Reply:
That's true, however it's also possible to put the same dialog elsewhere, such as in the page. Another solution is exposing a `onClickEditCampaign` prop and moving it to the page. Any idea best solution for right now?

Reply:
Oh I didn't consider that you could just use the same dialog over and over again. Let's leave it as it is for now and we can see once we have more cases what might be a good refactor (if any).